File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Base/PanoBaseViewController.m
//
//  PanoBaseViewController.m
//  PanoVideoCall
//
//  Created by pano on 2020/11/19.
//  Copyright © 2020 Pano. All rights reserved.
//

#import "PanoBaseViewController.h"
#import "UIImage+Extension.h"

@interface PanoBaseViewController ()

@end

@implementation PanoBaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"app_back"] style:UIBarButtonItemStylePlain target:self action:@selector(dismiss)];
    [self initViews];
    [self initConstraints];
    [self initService];
}

- (void)initViews {
    
}

- (void)initConstraints {
    
}

- (void)initService {
    
}

- (void)dismiss {
    [self dismissViewControllerAnimated:true completion:^{
    }];
}

- (void)dealloc {
    NSLog(@"%@ dealloc", self);
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/PanoBaseViewController.swift
//
//  PanoBaseViewController.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/10/29.
//

import UIKit

class PanoBaseViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.white;
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "app_back"), style: .plain, target: self, action: #selector(pop))
        initViews();
        initConstraints();
        initServices();
    }
    
    func initViews() {
        
    }
    
    func initConstraints() {
        
    }
    
    func initServices() {
        
    }
    
    func dismiss() {
        pop()
    }
    
    @objc func pop() {
        self.dismiss(animated: true, completion: nil)
//        else if self.navigationController != nil {
//            self.navigationController?.popViewController(animated: true)
//        }
    }
    
    func topSafeArea() -> CGFloat {
        var top = PanoStatusBarHeight()
        if self.navigationController != nil {
            top = top + 44;
        }
        return top
    }
    
    deinit {
        print("\(self) deinit")
    }

}
